### ISSUE TITLE
📘 Rapport au format pdf

### ISSUE THREAD

**Issue by MalcolmBrt** [adriens/temps-attente-streamlit#57]
Un rapport devra être réalisé selon les recommandations suivantes :

- Forme du rapport : police 12, interligne simple, limité à 20 pages (sans les annexes).
- Contenu du rapport :
une page de garde avec le titre de votre projet, votre nom, votre formation, votre tuteur
une introduction (présentation du contexte et des objectifs du projet) 
une description de votre travail (le corps du rapport) 
une conclusion 
des références bibliographiques (cf. https://bib.umontreal.ca/citer/comment-citer)
des annexes "si besoin"

Date limite de dépôt du rapport dans Moodle (UNC) : 17/11/2024 23h59

**Comment by MalcolmBrt** [comment]
On avance sur le rapport avec @morgancarre, une première version du rapport sera rendu dans la semaine @adriens

**Comment by adriens** [comment]
Yes ça roule 
Je vous ferai la relecture et les feedback 
Pour la vidéo de demo livraison estimée à quand?

**Comment by MalcolmBrt** [comment]
Dans la semaine d'après je pense, faut que je vois avec @morgancarre où il en est sur le diapo

**Comment by adriens** [comment]
> Dans la semaine d'après je pense, faut que je vois avec @morgancarre où il en est sur le diapo

Yes, car j'en ai besoin pour faire l'évaluation.

**Comment by morgancarre** [comment]
Voilà le rapport du projet @adriens : 
[Rapport Projet Tutoré Temps Attente OPT.pdf](https://github.com/user-attachments/files/17724109/Rapport.Projet.Tutore.Temps.Attente.OPT.pdf)

**Comment by adriens** [comment]
Donc j'en déduis que le périmètre fonctionnel est figé

**Comment by adriens** [comment]
![image](https://github.com/user-attachments/assets/ce14844c-37f2-4ff1-b82e-d63a2db6cc67)

C'est uniquement sous ma direction

**Comment by morgancarre** [comment]
> ![image](https://private-user-images.githubusercontent.com/5235127/385515739-ce14844c-37f2-4ff1-b82e-d63a2db6cc67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE0NTY5NTksIm5iZiI6MTczMTQ1NjY1OSwicGF0aCI6Ii81MjM1MTI3LzM4NTUxNTczOS1jZTE0ODQ0Yy0zN2YyLTRmZjEtYjgyZS1kNjNhMmRiNmNjNjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMDAxMDU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTVkYWJlNjNlMjczMDQ2YjgzNmYyMjY0Yzg4N2JiZmNkMWMxZjg3NzFlZTdlOThiOTU5YWI5YTIxYzE4NDYyOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.R5zLx8WTWIbbOngChIhdnANzv2XgLK_yFxu4X7OPCtk)
> 
> C'est uniquement sous ma direction

Ok je vais modifier

**Comment by adriens** [comment]
## Remerciements

> Nous tenons à remercier tout particulièrement Adrien Sales pour la proposition de ce
> projet captivant ainsi que pour le suivi constant et les précieux conseils qu'il nous a
> prodigués tout au long de son déroulement.

Rajoluter le fait que ces derniers étaient opérés en week-end et soirées

>Le mercredi 18 septembre, nous avons rencontré Adrien Sales et Michèle Barre
après un échange de mails. Cette rencontre a permis de clarifier le sujet du projet et
d’en poser les bases.

Préciser que c'est ce jour-là que le sujet a été pitché par Adrien, la méthodologie adoptée

## Roadmap

![image](https://github.com/user-attachments/assets/d7894519-0014-4181-93f7-8fe938447a4e)


Un gantt serait bien avec les screenshot des milestones GH

## Stack

> La stack que nous avons choisie pour ce projet est Streamlit en Python, s

... et Développée sur Ubuntu et podman pour conteneuriser


## MVP première version "fonctionnelle" (v0.1)

Screenshot de la milestone serait intéressant pour avoir l'idée des issues et des dates de delivery

> Après avoir signalé ce bug à Vinh Faucher, un membre de la DSI de l’OPT

Vinh, membre de l'équipe d'Adrien

## ⚠️ Streamlit cloud

- Il n'a pas été fait mention de cette partie que j'ai mise en oeuvre... ni de ce que cela apporte au projet.... et comment fonctionne ce service
- Un paragraphe détaillant la cinématique de la création d'issue, créeation de PR, revue de code par me soins, puis merge puis déploiement Streamlit cloud serait une bonne chose

**Comment by morgancarre** [comment]
> ## Remerciements
> > Nous tenons à remercier tout particulièrement Adrien Sales pour la proposition de ce
> > projet captivant ainsi que pour le suivi constant et les précieux conseils qu'il nous a
> > prodigués tout au long de son déroulement.
> 
> Rajoluter le fait que ces derniers étaient opérés en week-end et soirées
> 
> > Le mercredi 18 septembre, nous avons rencontré Adrien Sales et Michèle Barre
> > après un échange de mails. Cette rencontre a permis de clarifier le sujet du projet et
> > d’en poser les bases.
> 
> Préciser que c'est ce jour-là que le sujet a été pitché par Adrien, la méthodologie adoptée
> 
> ## Roadmap
> ![image](https://private-user-images.githubusercontent.com/5235127/385517440-d7894519-0014-4181-93f7-8fe938447a4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE0NTc2MTEsIm5iZiI6MTczMTQ1NzMxMSwicGF0aCI6Ii81MjM1MTI3LzM4NTUxNzQ0MC1kNzg5NDUxOS0wMDE0LTQxODEtOTNmNy04ZmU5Mzg0NDdhNGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMDAyMTUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDBkYmIyZDQ5NzA4NTliMmE0NjUyZWQwMTk2M2U1NTBhOGI3ODUyY2QyMDYzYjU5NzFmNzRlOTAyZjUwZjIxZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ApqRTcaghyzH0Dm-Dmkmzh4hZYoQHk2YTxFv6ylMIF8)
> 
> Un gantt serait bien avec les screenshot des milestones GH
> 
> ## Stack
> > La stack que nous avons choisie pour ce projet est Streamlit en Python, s
> 
> ... et Développée sur Ubuntu et podman pour conteneuriser
> 
> ## MVP première version "fonctionnelle" (v0.1)
> Screenshot de la milestone serait intéressant pour avoir l'idée des issues et des dates de delivery
> 
> > Après avoir signalé ce bug à Vinh Faucher, un membre de la DSI de l’OPT
> 
> Vinh, membre de l'équipe d'Adrien
> 
> ## ⚠️ Streamlit cloud
> * Il n'a pas été fait mention de cette partie que j'ai mise en oeuvre... ni de ce que cela apporte au projet.... et comment fonctionne ce service
> * Un paragraphe détaillant la cinématique de la création d'issue, créeation de PR, revue de code par me soins, puis merge puis déploiement Streamlit cloud serait une bonne chose

Ok merci pour le retour rapide on va faire les modifs 👍

**Comment by adriens** [comment]
yes, pour le support de prezz, utilisez [Reveal.](https://revealjs.com/)js svp

**Comment by morgancarre** [comment]
> Un gantt serait bien avec les screenshot des milestones GH

Je prend des screens des miletones mais pour la [🏆 v1.0 : Historique et auto-refresh](https://github.com/adriens/temps-attente-streamlit/milestone/4) elle est pas complété à 100% parce qu'il y a encore deux issues ouvertes , c'est pas grave ?

**Comment by adriens** [comment]
> elle est pas complété à 100% parce qu'il y a encore deux issues ouvertes , c'est pas grave ?

Non, c'est pas grave, c'est surtout pour mettre en avant la méthodologie...mais ça serait mieux que ce soit clean

**Comment by morgancarre** [comment]
Modifications faites ✅​  :
[Rapport Projet Tutoré Temps Attente OPT (1).pdf](https://github.com/user-attachments/files/17740576/Rapport.Projet.Tutore.Temps.Attente.OPT.1.pdf)

**Comment by adriens** [comment]
OK, je relis ça.
Pensez également au support video qui s'appuiera sur la prezz cf
- https://github.com/adriens/temps-attente-streamlit/issues/56

J'attendrai ces éléments pour réaliser mon évaluation

**Comment by adriens** [comment]
- 🆗 Top pour les screnshots des milestones
- ☝️ Il manque le schema d'architecture de la solution (app python, apigee, streamlit cloud)

**Comment by MalcolmBrt** [comment]
@adriens bonjour, est ce que c'est bien :
![image](https://github.com/user-attachments/assets/fafc256a-6e5b-4aaa-a761-d60f6e549ebd)

**Comment by adriens** [comment]
Yes, pour plus de clarté, tu pourrais privilégier : 

- Diagramme avec des logos de chaque solutionsur https://app.diagrams.net
- Garder uniquement APIGEE
- Préciser que l'utilisateur accès via une tablette un pc ou un telephone
- Générer une version animée du flow...laquelle pourra être injectée via un GIF animé  dans le slideshow cf https://github.com/adriens/temps-attente-streamlit/issues/56

**Comment by MalcolmBrt** [comment]
D'accord, je m'occupe de ces modifications.

> * Générer une version animée du flow...laquelle pourra être injectée via un GIF animé  dans le slideshow cf [📰 Support de présentation au format reveal.js #56](https://github.com/adriens/temps-attente-streamlit/issues/56)

Je n'ai pas bien compris le dernier point, aurais tu un exemple ?

**Comment by adriens** [comment]
Générer le schema d'archi puis l'exporter, par exemple pour sortir un truc du genre : 

![nodes24_architecture](https://github.com/user-attachments/assets/25bab9f3-aec4-4348-8b6d-100cbf3cd606)

Pour le slideshow : cf [`opt-nc/livre-or-collaborateurs-glia-slideshow`](https://github.com/opt-nc/livre-or-collaborateurs-glia-slideshow) : demo live : https://bit.ly/3RuNs1o

...le index.html de la prezz dans `/docs` de cette repo

**Comment by MalcolmBrt** [comment]
Voici une version améliorée animée et avec les images
![Schéma architecture drawio](https://github.com/user-attachments/assets/5f0e7737-381a-46aa-88ab-f1adf42520e1)

**Comment by adriens** [comment]
yes, bcp bcp mieux :clap: 
Reste juste à dropper le logo GCP/APIgee afin d'indiquer quel middleware est utilisé par l'OPT pour servir l'API et le portail d'API.

![image](https://github.com/user-attachments/assets/08e92c87-c3f2-40eb-9c08-7bea7e9361a5)


![image](https://github.com/user-attachments/assets/bdffa2e3-64e1-410b-9b77-95d626830e20)

**Comment by MalcolmBrt** [comment]
Comme ça ?
![Schéma architecture drawio](https://github.com/user-attachments/assets/bd87d1f7-2ca3-4050-b05a-88b85c13cfa7)

**Comment by adriens** [comment]
Impeccable !

**Comment by MalcolmBrt** [comment]
Si tout est bon, voici la version finale du rapport :
[Rapport Projet Tutoré Temps Attente OPT.pdf](https://github.com/user-attachments/files/17783813/Rapport.Projet.Tutore.Temps.Attente.OPT.pdf)

**Comment by adriens** [comment]
Tu penses que tu peux ajouter une photo de l'application entrain de tourner sur une tablette avec l'historique ?

**Comment by MalcolmBrt** [comment]
Ouai ok. Une photo de la tablette ou un screen ?

**Comment by adriens** [comment]
Genre une photo de la tablette posée sur un meuble qui met en évidence un chart d'historique de temps d'attente
